server.go

```diff
@@ -257,6 +257,8 @@ type server struct {
 
 	invoices *invoices.InvoiceRegistry
 
+	invoiceSettlementInterceptor *invoices.SettlementInterceptor
```

Comment:
same here re option pattern

Reply:
When I use the option pattern here I get a warning/error because `invoices.SettlementInterceptor` is a service which contains a field of type `sync.WaitGroup` that shouldn't be copied.